### ISSUE TITLE
BUILD, ENH: use codespell, flake8 to check code before merging

### ISSUE THREAD

**Issue by mattip** [numpy/numpy#13694]
In #13664 @luzpaz used codespell to find typos. We could maybe do this by adding this to a travis CI run. I wonder how many false positives it would generate. While we are at it, we could check new code similarily for flake8 violations.

`mapfile -t files < <(git diff --name-only $TRAVIS_COMMIT_RANGE); codespell -d -q 3 -S "$CODESPELL_SKIPS" -I ./whitelist.txt  "${files[@]}"`

**Comment by luzpaz** [comment]
@mattip FYI if there was a preference not to use a `whitelist.txt` file then one could use the `-L` flag with comma separated entries of whitelisted words.   

Also, other relevant numpy codespell PRs:  
#10212  
#10327  
#10604  
#10831  
#11005  
#10300

**Comment by ganesh-k13** [comment]
This is related to #18423, can we add codespell as another non-failing stage to GitHub actions?

**Comment by ganesh-k13** [comment]
@mattip I saw the latest triage meeting notes and there was an action item to make the linter non-blocking (In azure I guess as it does not block in actions), I can take it up in the same PR as this.

**Comment by seberg** [comment]
Yeah, that would be good. @charris is also looking into switching most of these over to 3.8 (but I guess that should be orthogonal).

**Comment by charris** [comment]
Already done in #19576, but it may not be quite right. I believe github/actions was the main problem, not sure if azure-pipelines is fixed.

**Comment by bsipocz** [comment]
Is adding codespell to CI still on the table? The PR (#19594) is now conflicting and probably easier to start from the beginning that rebasing and fixing.
